tomopy/misc/npmath.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import numpy  as np
+import logging
+import warnings
+
+logger = logging.getLogger(__name__)
+
+
+__author__ = "Chen Zhang"
+__all__    = [ 'gauss',
+               'calc_cummulative_dist',
+               'calc_affine_transform',
+]
+
+
+def gauss(x, *p):
+    """
+    simple Guassian func used for quick curve fit
+
+    Parameters
+    ----------
+    x  :  np.1ndarray
+        1D array for curve fitting
+    p  :  parameter lis t
+        A,          mu,      sigma  =   p
+        magnitude  center     std
+
+    Returns
+    -------
+    1d Gaussian distrobution evaluted at x with p
+    """
+    A, mu, sigma = p
+    return A*np.exp(-(x-mu)**2/(2.*sigma**2))
+
+
+def calc_cummulative_dist(data, steps=None):
+    """
+    Calculated the cumulative distribution from a data array without using
+    binning.  
+    This provides a more robust way to representing the statistics of the 
+    incoming data without worrying about the binning/samping effect.
+
+    Parameters
+    ----------
+    data  :  1D np.array
+        1-D numpy array 
+    steps :  [ None | int ], optional
+        Number of elements in the returning array
+
+    Returns
+    -------
+    pltX  : 1D np.array
+        plt data along x (data direction)
+    pltY  : 1D np.array
+        plt data long y (density direction)
+    """
+    x = np.sort(data)
+
+    # check if list empty
+    if len(x) < 1:
+        return [], []
+
+    # subsamping if steps is speficiied and the number is smaller than the
+    # total lenght of x
+    if (steps is not None) and len(x) > steps:
+        x = x[np.arange(0, len(x), int(np.ceil(len(x)/steps)))]
+
+    # calculate the cumulative density
+    xx = np.tile(x, (2, 1)).flatten(order='F')
+    y = np.arange(len(x))
+    yy = np.vstack((y, y+1)).flatten(order='F')/float(y[-1])
+
+    return xx, yy
+
+
+def calc_affine_transform(pts_src, pts_tgt):
```

Comment:
Why not use skimage.transform.AffineTransform instead?

Reply:
skimage.transform.AffineTransform() is used to transform an image with a given 2D affine transformation matrix.  This function generates the affine transformation matrix with given sets of points.

Reply:
Then, I'd rename the function something like affine_transform_matrix. Otherwise it is counter-intuitive.

Reply:
The prefix calc_  already indicates this is a calculation, not an execution.  Also using calc_ instead of get_ implies that this is not an o(1) calculation.  If I were to implement the execution part myself, I would probably name it "do_affine_transform" instead "affineTransform".